spec/master_spec.js

```diff
@@ -0,0 +1,172 @@
+    });
+
+    it("'specDone' should forward the payload directly", function() {
+      var env = { workerCount: 1 },
+          fakeJasmine = jasmine.createSpyObj('jasmine', ["execute", "configureDefaultReporter", "addSpecFiles"]);
+
+      fakeJasmine.specFiles = [];
+      fakeJasmine.reporter = jasmine.createSpyObj('reporter', ['specDone']);
+      runMasterJasmine(fakeJasmine, env, console.log);
+
+      var result = "jasmine";
+      cluster.emit("message", { send: noop, kill: noop }, { kind: "specDone", result: result });
+      expect(fakeJasmine.reporter.specDone).toHaveBeenCalledWith(result);
+    });
+
+    it("'suiteStarted' should forward the payload directly", function() {
+      var env = { workerCount: 1 },
+          fakeJasmine = jasmine.createSpyObj('jasmine', ["execute", "configureDefaultReporter", "addSpecFiles"]);
+
+      fakeJasmine.specFiles = [];
+      fakeJasmine.reporter = jasmine.createSpyObj('reporter', ['suiteStarted']);
+      runMasterJasmine(fakeJasmine, env, console.log);
+
+      var result = "jasmine";
+      cluster.emit("message", { send: noop, kill: noop }, { kind: "suiteStarted", result: result });
+      expect(fakeJasmine.reporter.suiteStarted).toHaveBeenCalledWith(result);
+    });
+
+    it("'suiteDone' should forward the payload directly", function() {
+      var env = { workerCount: 1 },
+          fakeJasmine = jasmine.createSpyObj('jasmine', ["execute", "configureDefaultReporter", "addSpecFiles"]);
+
+      fakeJasmine.specFiles = [];
+      fakeJasmine.reporter = jasmine.createSpyObj('reporter', ['suiteDone']);
+      runMasterJasmine(fakeJasmine, env, console.log);
+
+      var result = "jasmine";
+      cluster.emit("message", { send: noop, kill: noop }, { kind: "suiteDone", result: result });
+      expect(fakeJasmine.reporter.suiteDone).toHaveBeenCalledWith(result);
+    });
+
+    it("'jasmineStarted' should call once", function() {
+      var result = "passed",
+          fakeJasmine = jasmine.createSpyObj('jasmine', ["execute", "configureDefaultReporter", "addSpecFiles"]);
+
+      fakeJasmine.specFiles = [];
+      fakeJasmine.reporter = jasmine.createSpyObj('reporter', ['jasmineStarted']);
+      runMasterJasmine(fakeJasmine, {}, console.log);
+
+      cluster.emit("message", { send: noop, kill: noop }, { kind: "jasmineStarted", result: result });
+      expect(fakeJasmine.reporter.jasmineStarted).not.toBeDefined();
+
+      expect(function () {
+        cluster.emit("message", { send: noop, kill: noop }, { kind: "jasmineStarted", result: result });
+      }).not.toThrow();
+    });
+
+    it("'jasmineDone' should not be forwarded", function() {
+      var fakeJasmine = jasmine.createSpyObj('jasmine', ["execute", "configureDefaultReporter", "addSpecFiles"]);
+      
+      fakeJasmine.specFiles = [];
+      fakeJasmine.reporter = jasmine.createSpyObj('reporter', ['jasmineDone']);
+      runMasterJasmine(fakeJasmine, {}, console.log);
```

Comment:
I don't think we should really be passing in the real `console.log` here in test. I would at least expect some kind of mock printer that doesn't actually put things in the console.